### DISCUSSION TITLE
Import/Export

### DISCUSSION THREAD

**Discussion by jsmorabito** [athensresearch/athens#874] [Feature Requests]
- [ ] markdown
- [ ] json
- [ ] edn

which other file types would be good?

**Comment by jefftangx** [comment]
edn

**Reply by aleksandersumowski** [reply]
Is there a way to export roam research compatible EDN?

**Comment by Limezy** [comment]
Since Athens is based on a "closed" database, the ability to program an automatic export of all notes (for example, to .md format) would be great for resilience of all the knowledge

**Reply by jefftangx** [reply]
I agree. That should a setting IMO. "Every Y hrs create an .md backup at X location"

**Reply by jsmorabito** [reply]
Is it possible to make it so that editing the .md would also change the corresponding page in the .transit? i think that's how logseq does it but not sure... this is more related to #761 but something to consider

**Reply by Limezy** [reply]
Yes, that's how Logseq does indeed. That would be really great, but I don't know how it would scale up performance wise. And you can try Logseq, the .md file they are using as a database are actually quite far from readable anyway --> Even though the database is using .md files, you still have to export your pages as markdown to have a workable file.

Because of options such as the block quotes, I'm not really sure we could have the best of both worlds, that is a set of md files that can be both readable and used as the database (or synchronized real time to it, which is the same thing)

**Reply by jefftangx** [reply]
I think it's really cool that you can do that in LogSeq, but seems difficult to maintain. Not every feature will be markdown-compatible though. Initially there's block refs, but think about diagrams, boards, canvases, etc.

Export single page or entire graph definitely enough for now.

**Comment by panomitrius** [comment]
Obsidian-style md would be great.

**Comment by rjbergerud** [comment]
Being able to import markdown files would be useful for Polar Bookshelf users: after annotating, you can export annotations as markdown, and import to athens.

**Comment by japhir** [comment]
Exporting/importing to the [emacs](https://www.gnu.org/software/emacs/) [org-mode](https://orgmode.org/) file format would be great too. Moreover, it may be useful to use it as the main filetype! It has all your usual markup stuff, but also source code blocks, included LaTeX support, tables, etc.

And of course it's free software!

Org-mode supports:
- headings and subheadings are just a line starting with `*`, so `* heading 1`, `** subheading`
- lists (unordered/ordered)
- `- [ ] checkboxes` that can be `- [x] checked`
- TODO items (just headlines with configurable TODO keywords, such as NEXT WAITING DONE)
- deadlines/scheduled times, random timestamps in this format <2021-05-18 10:15>
  `#+deadline: <2021-05-20>`
- tables
  ```
  | heading | another heading |
  |---------|-----------------|
  |       5 |              18 |
  ```
- links/images
  `a link [[https://example.com][example]]`, and `[[file:image.png]]`
- source blocks in many programming languages (python, R, shell, ...) that can return images
  ```
  #+begin_src sh :results output
  echo "Hello world!"
  #+end_src

  #+results:
  : Hello world!
  ```
- references, footnotes, etc.

**Reply by japhir** [reply]
I've just seen the huge [list of features](https://athensresearch.github.io/athens/#/page/RTlsXxQ7Z) that you do not intend to implement, so I guess using org-mode as a backend without support for all of the above mentioned features would be counterproductive ;-).

**Reply by jefftangx** [reply]
@japhir it's unlikely we will support org-mode compatibility as first-class, but maybe with plugins one day. If that is important to you, I recommend org-roam or LogSeq!

**Reply by japhir** [reply]
Thanks @tangjeff0, I already use org-roam but was hoping one of these fancier apps would allow me to browse everything as nicely on my phone. [orgzly](http://www.orgzly.com/) is great but not for many files. LogSeq seems to have come a long way since the last time I looked at it, I love the drawing support they have, that's the only thing I'm missing from my plaintext life.

**Comment by Limezy** [comment]
As I posted in the GH issue, we should definitely be able to merge two Athens db. So the .transit format should be supported for import. Idealy, we should also be able to select page by page and then export to .transit.

**Comment by VictorVow** [comment]
OPML - used by WorkFlowy, Dynalist and various other outliners.

**Comment by latvia234** [comment]
Together with some more [basic keyboard shortcuts](https://github.com/athensresearch/athens/issues/565) this is the feature that I need the most.

Please add OPML, TXT/MD import and export per page.

Could you also add an option to include or exclude the - sign for every row?

**Comment by zhangym** [comment]
How to export to markdown or Roam Research file format？

**Reply by benwoodward** [reply]
I'd like this feature so that I can use Athens data to create a 'Digital Garden' using this tool: https://github.com/roam-garden/roam-export

**Comment by latvia234** [comment]
Related: https://github.com/shepheb/athens-export

**Comment by jialunzhang-psu** [comment]
Being able to export the whole database to a bunch of markdown files is quite important, gonna switch to Athens from Logseq once this feature is added. Athens's UI is much better

**Reply by JankinShaw** [reply]
Me too! Athens is much more beautiful and smoother than Logseq. It's just there isn't a way to export md files from Athens. Sadly

**Comment by latvia234** [comment]
All organizations in Europe have to comply with the GDPR. For us it is necessary to have a reliable export option. When we use personal data inside of an Athens database, we need the option to export it "in an accessible format". If someone asks our organization what kind of data we have from that person, it would be great if there is an option to export data.

Please read https://europa.eu/youreurope/business/dealing-with-customers/data-protection/data-protection-gdpr/index_en.htm under "Right to access and right to data portability"

It is great that we can host the Athens database ourselves, I think that most people in Europe really like this. But because of GDPR we also need the option to export the data. If this is not possible, European organizations cannot comply with the GDPR.

Let's hope that the Athens team will develop options to import/export right after RTC is completed.

**Comment by shivekkhurana** [comment]
Can you also consider exporting `html`? 

Athens could become my default blogging tool if I can export static HTML, then push it to a Github Repo and boom, got a free website.

**Comment by hydrogenbond007** [comment]
I was trying to export the Athens database to Roam but don't know how . I looked where the data was stored and its on .transit format. Could anyone help?
